### PR TITLE
Implement #else and #elif (without expressions)

### LINKED ISSUE #9: Support #elif and #else directives
They can be implemented without waiting for the full `#if` mini-language to be implemented.

### PR DESCRIPTION
**Pull request by Kangz**
Fixes #9

PTAL!

### CODE REVIEW

File: src/pp.rs [hunk 1]
```diff
@@ -333,6 +340,65 @@ impl<'a> DirectiveProcessor<'a> {
...
+            // The condition isn't parsed if it doesn't need to (and doesn't produce errors).
+            if block.outer_skipped || block.had_valid_segment {
+                return self.consume_until_newline();
+            }
+        } else {
+            return Err(StepExit::Error((
+                PreprocessorError::ElifOutsideOfBlock,
+                directive_location,
+            )));
+        };
```
Comment (pjoe) [inline src/pp.rs]: In the case where you have `if let Some(x) = y ... else return Err(z)` you can instead use:
```
let x = y.ok_or(z)?
```

See also: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or

File: src/pp.rs [hunk 2]
```diff
@@ -333,6 +340,65 @@ impl<'a> DirectiveProcessor<'a> {
...
+            todo!();
+        }
+
+        Ok(())
+    }
+
+    fn parse_else_directive(&mut self, directive_location: Location) -> Step<()> {
+        self.expect_lexer_token(LexerTokenValue::NewLine, directive_location)?;
+
+        if let Some(block) = self.blocks.last_mut() {
```
Comment (pjoe) [inline src/pp.rs]: try:
```rust
let mut block = self.blocks.last_mut().ok_or(StepExit::Error((
    PreprocessorError::MoreThanOneElse,
    directive_location,
)))?;
```

**Review by pjoe — APPROVED** [review]
lgtm 👍

**Comment by Kangz** [comment]
Thanks!

**Pull request merged**